docs/AdvancedContributorGuidelines.md

```diff
@@ -0,0 +1,48 @@
+# Advanced Contributor Guidelines
+
+## Choosing issues
+
+* Follow issue priorities. Prefer to take higher priority issues over lower priority issues.
+* Favor issues related to certain features/aspects/techs to deepen expertise in those areas. 
+ 
+## Creating commits in big PRs
+
+Big PRs are hard to review. More systematic committing (as explained below) can ease the workload of the reviewer.
+ 
+* Plan your PR as a sequence of steps, each step advancing the code base in a logical step towards 
+  the end goal of the PR. The build should pass at each step.
+* Commit each step as a separate commit. You may have to `squash` trivial commits or reorder commits to make the steps
+  more logical. In the end, the series of commits should match the exact sequence of steps.
+* Write more detailed commit messages as if you were explaining your commit to a future developer.
+* Post a summary of your commits in the PR description so that the reviewer can refer to them easily in review comments.
+  Here is an example (adapted from [this PR](https://github.com/CS2103AUG2016-T11-C4/main/pull/120)). 
+  
+  ```
+  Rework tests so that they do not depend on new Task().
+  
+  [1/4] TaskBook::resetData(): replace floating/event/deadline task data as well
+  [2/4] logic: remove non-TaskTracker related commands
+  [3/4] testutil: remove SerializableTestClass
+  [4/4] testutil: remove TaskBookBuilder
+  ```
+  
+  > TODO : add a link to the page for generating PR summary
```

Comment:
@pyokagan 

> I'm not supporting or maintaining it though, so I think you will need to set up an actual project to maintain the supporting tools for this workflow.

I'll use this one for now. We can fork it to oss-generic later.